src/pages/product-details/comments/comments.js

```diff
@@ -0,0 +1,77 @@
+import React, { useState } from 'react';
+import { useSelector } from 'react-redux';
+
+import Rating from '@material-ui/lab/Rating';
+import TextField from '@material-ui/core/TextField';
+import Button from '@material-ui/core/Button';
+import useStyles from './comments.styles';
+
+import CommentsItem from './comments-item';
+
+import { INPUT_VARIANT } from '../../../configs';
+import { COMMENTS } from '../../../translations/product-details.translations';
+
+const Comments = () => {
+  const styles = useStyles();
+  const { language, comments } = useSelector(({ Language, Products }) => ({
+    language: Language.language,
+    comments: Products.product.comments
+  }));
+
+  const [rate, setRate] = useState(0);
```

Comment:
It should be handled inside the `redux`

Reply:
It's local Material UI state for rate, so when user change the rate render of it changed. I don't think, that we need every time dispatch it to the store, when user change locally rate, that he wants to leave